drivers/flash/nrf_qspi_nor.c

```diff
@@ -487,16 +487,19 @@ static inline int qspi_nor_read_id(struct device *dev,
 static int qspi_nor_read(struct device *dev, off_t addr, void *dest,
 			 size_t size)
 {
+	void *dptr = dest;
+	size_t dlen = size;
+	u8_t buf[4];
+
 	if (!dest) {
 		return -EINVAL;
 	}
 
 	/* read size must be non-zero multiple of 4 bytes */
-	if (((size % 4U) != 0) || (size == 0)) {
-		return -EINVAL;
-	}
-	/* address must be 4-byte aligned */
-	if ((addr % 4U) != 0) {
+	if (size < 4U) {
+		dest = buf;
```

Comment:
I do not know. I find it counterintuitive that `*dptr` local variable gets defined and assigned `dest` function parameter, so that the parameter could be modified.

Reply:
That change if from upstream so I can't really do anything about that so I just copied the way it is upstream for write also.

Reply:
Ok then. My bad.

Reply:
Actually if you have some spare time, could you echange the `dest` and `dptr`, you are working on the code anyway.